data_structures/stacks/balanced_parentheses.py

```diff
@@ -1,23 +1,35 @@
 from .stack import Stack
 
-__author__ = "Omkar Pathak"
 
-
-def balanced_parentheses(parentheses):
-    """ Use a stack to check if a string of parentheses is balanced."""
-    stack = Stack(len(parentheses))
-    for parenthesis in parentheses:
-        if parenthesis == "(":
-            stack.push(parenthesis)
-        elif parenthesis == ")":
-            if stack.is_empty():
+def balanced_parentheses(parentheses: str) -> bool:
+    """Use a stack to check if a string of parentheses is balanced.
+    >>> balanced_parentheses("([]{})")
+    True
+    >>> balanced_parentheses("[()]{}{[()()]()}")
+    True
+    >>> balanced_parentheses("[(])")
+    False
```

Comment:
Please add a string that has no brackets and also an empty string.

Reply:
Done. Please review again.